src/core/server/http/base_path_service.ts

```diff
@@ -22,13 +22,15 @@ import { modifyUrl } from '../../utils';
 
 export class BasePath {
   private readonly basePathCache = new WeakMap<LegacyRequest, string>();
+  public readonly serverBasePath: string;
 
-  constructor(private readonly serverBasePath?: string) {}
+  constructor(serverBasePath?: string) {
+    this.serverBasePath = serverBasePath || '';
+  }
```

Comment:
Does this not work?

```ts
constructor(public readonly serverBasePath = '') {}
```

Reply:
That would be simpler, but I had to split it out to document the public property.